### PR TITLE
Add checks for missing stations

### PR DESCRIPTION
**Pull request by conbrad**
The pdf endpoint assumes the list of stations from WF1 is the same list of stations for a HFI request. This is not true since we made the change to render the HFI table based on the request. 

The calculations are based on the HFI request as well so now every function/artifact is based on it.

Tested with data from prod.
# Test Links:
[Landing Page](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2838.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Comment by andrea-williams** [comment]
I've found an issue, but I'm not sure if it's related to this bugfix or something completely separate.

1. in the dev env for this PR, select Southeast FC with default prep period for today (May 9 - May 13)
2. for East Kootenay Zone, deselect the 2 stations that don't have any weather data available
3. the 4 other stations in that planning area appear to be reporting all necessary weather and fire index data, but HFI Calc is showing an error saying that the mean intensity group can't be calculated (but it should be as straightforward as calculating the mean from the intensity groups already shown)
4. the same issue is carried on into the PDF

Screenshot 2023-05-09 at 1 51 32 PM
[SoutheastFireCentre_HFICalculator_2023-05-09_AWILLIAM.pdf](https://github.com/bcgov/wps/files/11436120/SoutheastFireCentre_HFICalculator_2023-05-09_AWILLIAM.pdf)

File: api/app/hfi/pdf_data_formatter.py [hunk 1]
```diff
@@ -140,16 +140,21 @@ def get_merged_station_data(
...
     all_station_pdf_data: List[StationPDFData] = []
+    # We do a bunch of null checks here because station_dict is built from
+    # stations we retrieve from WF1, while the dailies come from the HFI request
+    # which can have an different set of stations
     for daily in dailies:
-        station_data = station_dict[daily.code]
+        station_data = station_dict.get(daily.code, None)
         daily_dict = daily.dict()
-        daily_dict.update(station_data)
+        if station_data is not None:
```
Comment (dgboss) [inline api/app/hfi/pdf_data_formatter.py:149]: There are two null checks for station_data. Is it possible to consolidate?

Comment (dgboss) [reply api/app/hfi/pdf_data_formatter.py:149]: Perhaps with 
```
if station_data is None:
    continue
```
at the first ocurrence of the null check and then remove the second null check?

**Review by andrea-williams — APPROVED** [review]
Have created new bug ticket for previously mentioned bug.

Code LGTM - I'm not sure how exactly to test the functionality in dev.

File: api/app/hfi/pdf_data_formatter.py (hunk 1 above)
Comment (conbrad) [reply api/app/hfi/pdf_data_formatter.py]: Good call, done in: https://github.com/bcgov/wps/pull/2838/commits/2b7eae72cc6d9b22bff392ade68cae24efcf5b5f

**Pull request merged**